tools/samtools_bam_to_cram.cwl

```diff
@@ -0,0 +1,32 @@
+cwlVersion: v1.0
+class: CommandLineTool
+id: samtools_bam_to_cram
+doc: |-
+  This tool converts the input BAM into a CRAM.
+  The following programs are run in this tool:
+    - samtools view
+    - samtools index
+requirements:
+  - class: ShellCommandRequirement
+  - class: InlineJavascriptRequirement
+  - class: ResourceRequirement
+    ramMin: 4000
+  - class: DockerRequirement
+    dockerPull: 'pgc-images.sbgenomics.com/d3b-bixu/samtools:1.9'
+baseCommand: [samtools, view]
+arguments:
+  - position: 1
+    shellQuote: false
+    valueFrom: >-
+      -C -o $(inputs.input_bam.nameroot).cram
+  - position: 2
+    shellQuote: false
+    valueFrom: >-
+      && samtools index $(inputs.input_bam.nameroot).cram
+inputs:
+  reference: {type: File, secondaryFiles: [.fai], doc: "Reference fasta with associated fai index",
+    inputBinding: {prefix: "-T", position: 1 }}
+  input_bam: {type: File, secondaryFiles: [^.bai], doc: "Input bam file",
+    inputBinding: { position: 1 }}
```

Comment:
It's supposed to be the last thing in the samtools view command. Probably want to give it a position that denotes that. You'd have to bump the second argument as well.

Reply:
So, I could be wrong, but I think it has tie-breaking rules. If they are all position 1 then:
1. stuff written in `arguments` is eval/positions first
2. If `inputs` are both assigned to the same position, it just goes in order in which the input was specified. So, since `input_bam` was given last, it should show up last. Having said all that, probably cleaner to just increment the position numbers to make it clear to the user that this is what you wanted.